File: data/java/tests/test_spacy_pos.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_spacy_pos {

    @Test
    void testGetPosTagsEmptyText() {
        function_spacy_pos processor = new function_spacy_pos();
        String result = processor.test_pos("");
        assertTrue(result instanceof String);
    }

    @Test
    void testNounVerbAdjectiveAdverbPunctuation() {
        function_spacy_pos processor = new function_spacy_pos();
        String text = "The quick brown fox jumps quickly. And it's a good day!";
        String result = processor.test_pos(text);

        // Print actual output for debugging
        System.out.println("Actual output: " + result);

        // 使用contains而不是完全匹配，因为单词的顺序可能不同
        assertTrue(result.contains("fox (NOUN)"), "Should contain 'fox (NOUN)'");
        assertTrue(result.contains("day (NOUN)"), "Should contain 'day (NOUN)'");
        assertTrue(result.contains("jumps (VERB)"), "Should contain 'jumps (VERB)'");
        assertTrue(result.contains("quick (ADJ)"), "Should contain 'quick (ADJ)'");
        assertTrue(result.contains("good (ADJ)"), "Should contain 'good (ADJ)'");
        assertTrue(result.contains("quickly (ADV)"), "Should contain 'quickly (ADV)'");
        assertTrue(result.contains(". (PUNCT)"), "Should contain '. (PUNCT)'");
        assertTrue(result.contains("! (PUNCT)"), "Should contain '! (PUNCT)'");
    }

    @Test
    void testGetPosTagsSentence() {
        function_spacy_pos processor = new function_spacy_pos();
        String text = "The quick brown fox jumps over the lazy dog.";
        String result = processor.test_pos(text);
        assertEquals(10, result.split(", ").length);
    }

    @Test
    void testGetPosTagsSentenceType() {
        function_spacy_pos processor = new function_spacy_pos();
        String text = "The quick brown fox jumps over the lazy dog.";
        String result = processor.test_pos(text);

        for (String wordPos : result.split(", ")) {
            String[] parts = wordPos.split(" ");
            assertTrue(parts[0] instanceof String);
            assertTrue(parts[1] instanceof String);
        }
    }

    @Test
    void testGetPosTagsNonStringInput() {
        function_spacy_pos processor = new function_spacy_pos();
        assertThrows(IllegalArgumentException.class, () -> {
            processor.test_pos(null);
        });
    }
}

File: code/test_suites/python/python/test_spacy_pos.py
import unittest
from function_spacy_pos import SpacyPosProcessor


class TestGetPosTags(unittest.TestCase):

    def test_get_pos_tags_empty_text(self):
        processor = SpacyPosProcessor()
        result = processor.test_pos("")
        self.assertEqual(type(result), str)  # 确保返回结果是字符串

    def test_noun_verb_adjective_adverb_punctuation(self):
        processor = SpacyPosProcessor()
        text = "The quick brown fox jumps quickly. And it's a good day!"
        result = processor.test_pos(text)

        # 验证每个单词和词性
        self.assertIn("fox (NOUN)", result)
        self.assertIn("day (NOUN)", result)
        self.assertIn("jumps (VERB)", result)
        self.assertIn("quick (ADJ)", result)
        self.assertIn("good (ADJ)", result)
        self.assertIn("quickly (ADV)", result)
        self.assertIn(". (PUNCT)", result)
        self.assertIn("! (PUNCT)", result)

    def test_get_pos_tags_sentence(self):
        processor = SpacyPosProcessor()
        text = "The quick brown fox jumps over the lazy dog."
        result = processor.test_pos(text)
        self.assertEqual(len(result.split(", ")), 10)  # 确保返回的词性标注数量为10

    def test_get_pos_tags_sentence_type(self):
        processor = SpacyPosProcessor()
        text = "The quick brown fox jumps over the lazy dog."
        result = processor.test_pos(text)
        # 确保每个词和词性都是字符串类型
        for word_pos in result.split(", "):
            word, pos = word_pos.split(" ")
            self.assertEqual(type(word), str)
            self.assertEqual(type(pos), str)

    def test_get_pos_tags_non_string_input(self):
        processor = SpacyPosProcessor()
        with self.assertRaises(TypeError):
            processor.test_pos(123)  # 输入非字符串时抛出异常


if __name__ == '__main__':
    unittest.main()
